source/material_model/rheology/peierls_creep.cc

```diff
@@ -0,0 +1,229 @@
+               * q is the second Peierls creep fitting exponent
+               * A is the Peierls prefactor term (1/s 1/Pa^2)
+               * n is the Peierls stress exponent
+               * E is the Peierls activation energy (J/mol)
+               * V is the Peierls activation volume (m^3/mol)
+               * edot_ii is the second invariant of the deviatoric strain rate (1/s)
+               * T is temperature (K)
+               * R is the gas constant
+               */
+              const double s = ( (activation_energies_peierls[c] + pressure * activation_volumes_peierls[c]) / (constants::gas_constant * temperature)) *
+                               peierls_fitting_exponents_p[c] * peierls_fitting_exponents_q[c] *
+                               std::pow((1. - std::pow(peierls_fitting_parameters[c], peierls_fitting_exponents_p[c])),(peierls_fitting_exponents_q[c] - 1.0)) *
+                               std::pow(peierls_fitting_parameters[c], peierls_fitting_exponents_p[c]);
+
+              const double peierls_left_term = 0.5 * (peierls_fitting_parameters[c] * peierls_stresses[c]) /
+                                               std::pow((prefactors_peierls[c] * std::pow(peierls_fitting_parameters[c] * peierls_stresses[c],stress_exponents_peierls[c])),( 1 / (s + stress_exponents_peierls[c])));
+
+              const double peierls_middle_term_a = (activation_energies_peierls[c] + pressure * activation_volumes_peierls[c]) / (constants::gas_constant * temperature);
+
+              const double peierls_middle_term_b = std::pow((1. - std::pow(peierls_fitting_parameters[c],peierls_fitting_exponents_p[c])),peierls_fitting_exponents_q[c]);
+
+              const double peierls_middle_term_c = s + stress_exponents_peierls[c];
+
+              const double peierls_middle_term = std::exp(peierls_middle_term_a * peierls_middle_term_b / peierls_middle_term_c);
+
+              const double peierls_right_term = std::pow(strain_rate, (1. / (s + stress_exponents_peierls[c])) - 1.);
+
+              viscosity = peierls_left_term * peierls_middle_term * peierls_right_term;
+              break;
+            }
+            default:
+            {
+              AssertThrow(false, ExcNotImplemented());
+              break;
+            }
+          }
+
+        return viscosity;
+      }
+
+
+
+      template <int dim>
+      void
+      PeierlsCreep<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.declare_entry ("Peierls creep flow law", "viscosity approximation",
+                           Patterns::Selection("viscosity approximation"),
+                           "Select what type of Peierls creep flow law to use. Currently, the "
+                           "only available option is an approximation to Peierls creep, which "
+                           "uses the strain rate invariant, rather than stress, as an input. "
+                           "Future options will allow formulations that use stress as an input.");
```

Comment:
Never mention future things in code ... you dont know if it will happen next week, next year or never and in all cases the mention is not helpful at the moment, except to appease reviewers who would complain about a parameter that has only a single option ;-).

Reply:
Removed :-) I'll just open an issue as a reminder.